datadog_checks_base/datadog_checks/base/utils/tls.py

```diff
@@ -35,16 +35,24 @@
 class TlsContextWrapper(object):
     __slots__ = ('logger', 'config', 'tls_context')
 
-    def __init__(self, instance, remapper=None):
-        # type: (InstanceType, Dict[AnyStr, Dict[AnyStr, Any]]) -> None
+    def __init__(self, instance, remapper=None, overrides=None):
+        # type: (InstanceType, Dict[AnyStr, Dict[AnyStr, Any]], Dict[AnyStr, Any]) -> None
         default_fields = dict(STANDARD_FIELDS)
 
+        # Override existing config options if there exists any overrides
+        overridden_instance = {field: data for field, data in iteritems(instance)}
```

Comment:
Use deepcopy(instance)
Otherwise you might still be modifying instance behind the scenes